assets/dev/js/editor/regions/responsive-bar/view.js

```diff
@@ -122,11 +123,14 @@ export default class View extends Marionette.ItemView {
 	}
 
 	onBreakpointSelected( e ) {
-		const currentDeviceMode = elementor.channels.deviceMode.request( 'currentMode' ),
-			selectedDeviceMode = e.target.value;
+		const selectedDeviceMode = e.target.value;
+
+		elementor.changeDeviceMode( selectedDeviceMode, false );
```

Comment:
I deleted the if statment because it's always true. The function will never be triggered when you click on the same breakpoint.